### PR TITLE
add aggregate and aggregate snapshot modules

### PR DESCRIPTION
**Pull request by MohamedFarouk-HMCTS**

### CODE REVIEW

File: aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    SnapshotRepository snapshotRepository;
+
+    @Inject
+    SnapshotStrategy snapshotStrategy;
+
+    public void setStreamStrategy(ObjectInputStreamStrategy streamStrategy) {
+        this.streamStrategy = streamStrategy;
+    }
+
+    ObjectInputStreamStrategy streamStrategy = new DefaultObjectInputStreamStrategy();
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java]: Can this be private? 
Should be moved above any methods

File: aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    @Inject
+    Logger logger;
+
+    @Inject
+    SnapshotRepository snapshotRepository;
+
+    @Inject
+    SnapshotStrategy snapshotStrategy;
+
+    public void setStreamStrategy(ObjectInputStreamStrategy streamStrategy) {
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java]: final

File: aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }
+
+    @Override
+    public <T extends Aggregate> void removeAllSnapshots(final UUID streamId, final Class<T> clazz) {
+        logger.trace("Removing all snapshots for {}", streamId, clazz);
+        snapshotRepository.removeAllSnapshots(streamId, clazz);
+    }
+
+    @Override
+    public <T extends Aggregate> long getLatestSnapshotVersion(UUID streamId, Class<T> clazz) {
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotService.java]: finals

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/domain/aggregate/classloader/DynamicAggregateTestClassGenerator.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+import com.squareup.javapoet.FieldSpec;
+import com.squareup.javapoet.JavaFile;
+import com.squareup.javapoet.MethodSpec;
+import com.squareup.javapoet.ParameterizedTypeName;
+import com.squareup.javapoet.TypeName;
+import com.squareup.javapoet.TypeSpec;
+
+public class DynamicAggregateTestClassGenerator {
+
+    public <T extends Aggregate> Class<T> generatedTestAggregateClassOf(long serialVersionUid, String basePackage, String fileNameToBeGenerated) throws IOException, ClassNotFoundException {
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/domain/aggregate/classloader/DynamicAggregateTestClassGenerator.java]: Ideally could be finals in this class

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java [hunk 5]
```diff
@@ -0,0 +1,443 @@
...
+        MatcherAssert.assertThat(snapshot, IsNot.not(nullValue()));
+        MatcherAssert.assertThat(snapshot.isPresent(), equalTo(true));
+        MatcherAssert.assertThat(snapshot.get().getType(), equalTo(TYPE));
+        MatcherAssert.assertThat(snapshot.get().getStreamId(), equalTo(STREAM_ID));
+        MatcherAssert.assertThat(snapshot.get().getVersionId(), equalTo(25L));
+
+        MatcherAssert.assertThat(snapshotRepository.snapshotCount(STREAM_ID), is(1L));
+        MatcherAssert.assertThat(eventLogRepository.eventLogCount(STREAM_ID), is(25));
+        MatcherAssert.assertThat(aggregateFromSnapshot.numberOfAppliedEvents(), is(25));
+        MatcherAssert.assertThat(aggregateFromSnapshot.recordedEvents().size(), is(25));
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: static import could be used here

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java [hunk 6]
```diff
@@ -0,0 +1,443 @@
...
+        MatcherAssert.assertThat(snapshotChanged.isPresent(), equalTo(true));
+        MatcherAssert.assertThat(snapshotChanged.get().getType(), equalTo(aggregateClass.getName()));
+        MatcherAssert.assertThat(snapshotChanged.get().getStreamId(), equalTo(STREAM_ID));
+        MatcherAssert.assertThat(snapshotChanged.get().getVersionId(), equalTo(25L));
+
+        MatcherAssert.assertThat(snapshotRepository.snapshotCount(STREAM_ID), is(1L));
+        MatcherAssert.assertThat(eventLogRepository.eventLogCount(STREAM_ID), is(48));
+        TestAggregate aggregateFromSnapshot = snapshotChanged.get().getAggregate(new DefaultObjectInputStreamStrategy());
+        MatcherAssert.assertThat(aggregateFromSnapshot.numberOfAppliedEvents(), is(25));
+//        MatcherAssert.assertThat(aggregateFromSnapshot.recordedEvents(), is(getRecordedEvents(25)));
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: this commented code should be removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java [hunk 7]
```diff
@@ -0,0 +1,443 @@
...
+        MatcherAssert.assertThat(newSnapshot, IsNot.not(nullValue()));
+        MatcherAssert.assertThat(newSnapshot.isPresent(), equalTo(true));
+        MatcherAssert.assertThat(newSnapshot.get().getType(), equalTo(aggregateClass.getName()));
+        MatcherAssert.assertThat(newSnapshot.get().getStreamId(), equalTo(STREAM_ID));
+        MatcherAssert.assertThat(newSnapshot.get().getVersionId(), equalTo(initialNumberOfSnapshots * SNAPSHOT_THRESHOLD));
+        MatcherAssert.assertThat(snapshotRepository.snapshotCount(STREAM_ID), is(4L));
+        MatcherAssert.assertThat(eventLogRepository.eventLogCount(STREAM_ID), is(123));
+        TestAggregate aggregateFromSnapshot2 = (TestAggregate) newSnapshot.get().getAggregate(new DefaultObjectInputStreamStrategy());
+        MatcherAssert.assertThat(aggregateFromSnapshot2.numberOfAppliedEvents(), is(100));
+//        MatcherAssert.assertThat(aggregateFromSnapshot2.recordedEvents(), is(getRecordedEvents(100)));
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: can be removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java [hunk 8]
```diff
@@ -0,0 +1,443 @@
...
+        MatcherAssert.assertThat(snapshot.isPresent(), equalTo(true));
+        MatcherAssert.assertThat(snapshot.get().getType(), equalTo(TYPE));
+        MatcherAssert.assertThat(snapshot.get().getStreamId(), equalTo(STREAM_ID));
+        MatcherAssert.assertThat(snapshot.get().getVersionId(), equalTo(25L));
+        MatcherAssert.assertThat(snapshotRepository.snapshotCount(STREAM_ID), is(1L));
+        MatcherAssert.assertThat(eventLogRepository.eventLogCount(STREAM_ID), is(48));
+
+        TestAggregate aggregateFromSnapshot = snapshot.get().getAggregate(new DefaultObjectInputStreamStrategy());
+        MatcherAssert.assertThat(aggregateFromSnapshot.numberOfAppliedEvents(), is(25));
+//        MatcherAssert.assertThat(aggregateFromSnapshot.recordedEvents(), is(getRecordedEvents(25)));
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: needs to be removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java [hunk 9]
```diff
@@ -0,0 +1,443 @@
...
+        MatcherAssert.assertThat(snapshot.isPresent(), equalTo(true));
+        MatcherAssert.assertThat(snapshot.get().getType(), equalTo(TYPE));
+        MatcherAssert.assertThat(snapshot.get().getStreamId(), equalTo(STREAM_ID));
+        MatcherAssert.assertThat(snapshot.get().getVersionId(), equalTo(50L));
+        MatcherAssert.assertThat(snapshotRepository.snapshotCount(STREAM_ID), is(2L));
+        MatcherAssert.assertThat(eventLogRepository.eventLogCount(STREAM_ID), is(50));
+
+        TestAggregate aggregateFromSnapshot = (TestAggregate) snapshot.get().getAggregate(new DefaultObjectInputStreamStrategy());
+        MatcherAssert.assertThat(aggregateFromSnapshot.numberOfAppliedEvents(), is(50));
+//        MatcherAssert.assertThat(aggregateFromSnapshot.recordedEvents(), is(getRecordedEvents(50)));
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: remove

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java [hunk 10]
```diff
@@ -0,0 +1,443 @@
...
+        final Optional<AggregateSnapshot> newSnapshot = snapshotRepository.getLatestSnapshot(STREAM_ID, newAggregateClass);
+        MatcherAssert.assertThat(newSnapshot, IsNot.not(Matchers.nullValue()));
+        MatcherAssert.assertThat(newSnapshot.isPresent(), equalTo(true));
+        MatcherAssert.assertThat(newSnapshot.get().getType(), equalTo(newAggregateClass.getName()));
+        MatcherAssert.assertThat(newSnapshot.get().getStreamId(), equalTo(STREAM_ID));
+        MatcherAssert.assertThat(newSnapshot.get().getVersionId(), equalTo(123L));
+        MatcherAssert.assertThat(snapshotRepository.snapshotCount(STREAM_ID), is(1L));
+        MatcherAssert.assertThat(eventLogRepository.eventLogCount(STREAM_ID), is(123));
+
+        // MatcherAssert.assertThat(snapshot.get().getAggregate(new DefaultObjectInputStreamStrategy()).count(),is(100));
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: remove

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java [hunk 11]
```diff
@@ -0,0 +1,264 @@
...
+        aggregateService.get(eventStream, TestAggregate.class);
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void shouldThrowExceptionForNonInstantiatableEvent() throws AggregateChangeDetectedException {
+        defaultAggregateService.logger = logger;
+        defaultAggregateService.jsonObjectToObjectConverter = jsonObjectToObjectConverter;
+
+        when(snapshotService.getLatestVersionedAggregate(STREAM_ID, TestAggregate.class)).thenReturn(Optional.of(new VersionedAggregate<>(INITIAL_AGGREGATE_VERSION, new TestAggregate())));
+        JsonObject eventPayloadA = mock(JsonObject.class);
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java]: This class seems inconsistent some methods have finals some do not

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java [hunk 12]
```diff
@@ -0,0 +1,264 @@
...
+        final EventC eventC = new EventC("C1");
+
+        Mockito.when(eventStream.getId()).thenReturn(streamId);
+        Mockito.when(eventStream.getCurrentVersion()).thenReturn(currentStreamVersion);
+        Mockito.when(snapshotService.getLatestVersionedAggregate(streamId, TestAggregate.class)).thenReturn(
+                Optional.of(new VersionedAggregate<>(snapshotVersion, aggregate)));
+        Mockito.when(eventStream.readFrom(snapshotVersion + 1)).thenReturn(of(jsonEventA, jsonEventB, jsonEventC));
+
+
+        Mockito.when(jsonObjectToObjectConverter.convert(jsonEventA.payloadAsJsonObject(), EventA.class)).thenReturn(eventA);
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java]: static import

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+    @InjectMocks
+    private DefaultSnapshotService snapshotService;
+
+
+    @Test
+    public void shouldCreateAggregateWhenLatestVersionRequested() throws AggregateChangeDetectedException {
+        final Optional<AggregateSnapshot<TestAggregate>> aggregateSnapshot = Optional.empty();
+        when(snapshotRepository.getLatestSnapshot(STREAM_ID, TestAggregate.class)).thenReturn(aggregateSnapshot);
+
+        Optional<VersionedAggregate<TestAggregate>> versionedAggregate = snapshotService.getLatestVersionedAggregate(STREAM_ID, TestAggregate.class);
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java]: final missing

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+        assertThat(snapshotArgumentCaptor.getValue(), notNullValue());
+        assertThat(snapshotArgumentCaptor.getValue().getVersionId(), is(currentAggregateVersionId));
+
+    }
+
+    @Test
+    public void shouldNotCreateAggregateIfStrategyDoesNotMandatesCreation() throws AggregateChangeDetectedException {
+        final Long initialAggregateVersionId = 0l;
+        final Long currentAggregateVersionId = 26l;
+        TestAggregate aggregate = new TestAggregate();
```
Comment (mapingo) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java]: final

File: aggregate/pom.xml [hunk 15]
```diff
@@ -0,0 +1,84 @@
...
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>event-source</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>javax</groupId>
+            <artifactId>javaee-api</artifactId>
```
Comment (mapingo) [inline aggregate/pom.xml]: Should have <scope>provided</scope>
Should be at the first in the list

File: aggregate/pom.xml [hunk 16]
```diff
@@ -0,0 +1,84 @@
...
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>event-stream</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>event-source</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
```
Comment (mapingo) [inline aggregate/pom.xml]: This pom needs to be rearranged into correct order

File: aggregate/src/main/java/uk/gov/justice/services/core/aggregate/exception/AggregateCreationException.java [hunk 17]
```diff
@@ -0,0 +1,8 @@
+package uk.gov.justice.services.core.aggregate.exception;
+
+public class AggregateCreationException extends RuntimeException {
```
Comment (mapingo) [inline aggregate/src/main/java/uk/gov/justice/services/core/aggregate/exception/AggregateCreationException.java]: This seems to be not used

File: aggregate/src/test/java/uk/gov/justice/services/core/aggregate/DefaultAggregateServiceIT.java [hunk 18]
```diff
@@ -0,0 +1,242 @@
...
+        assertThat(aggregate.recordedEvents(), empty());
+    }
+
+    @Test
+    public void shouldCreateAggregateFromSingletonStream() throws EventStreamException {
+
+        final EventStream eventStream = eventSource.getStreamById(STREAM_ID);
+        aggregateService.register(new EventFoundEvent(EventA.class, "context.eventA"));
+
+        TestAggregate aggregate = aggregateService.get(eventStream, TestAggregate.class);
```
Comment (mapingo) [inline aggregate/src/test/java/uk/gov/justice/services/core/aggregate/DefaultAggregateServiceIT.java]: class needs to be consistent no final here and below

File: aggregate/src/test/java/uk/gov/justice/services/core/aggregate/event/EventA.java [hunk 19]
```diff
@@ -0,0 +1,22 @@
+package uk.gov.justice.services.core.aggregate.event;
+
+import uk.gov.justice.domain.annotation.Event;
+
+@Event("eventA")
+public class EventA {
```
Comment (mapingo) [inline aggregate/src/test/java/uk/gov/justice/services/core/aggregate/event/EventA.java]: Class seems to be not used

File: aggregate/src/test/java/uk/gov/justice/services/core/aggregate/event/EventB.java [hunk 20]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.justice.services.core.aggregate.event;
+
+import uk.gov.justice.domain.annotation.Event;
+
+@Event("eventB")
+public class EventB {
```
Comment (mapingo) [inline aggregate/src/test/java/uk/gov/justice/services/core/aggregate/event/EventB.java]: Class seems to be not used

File: aggregate/src/test/java/uk/gov/justice/services/core/aggregate/event/EventC.java [hunk 21]
```diff
@@ -0,0 +1,23 @@
+package uk.gov.justice.services.core.aggregate.event;
+
+import uk.gov.justice.domain.annotation.Event;
+
+@Event("eventC")
+public class EventC {
```
Comment (mapingo) [inline aggregate/src/test/java/uk/gov/justice/services/core/aggregate/event/EventC.java]: Class seems to be not used

File: aggregate-snapshot/aggregate-snapshot-service/pom.xml [hunk 22]
```diff
@@ -0,0 +1,157 @@
...
+        </dependency>
+        <dependency>
+            <groupId>uk.gov.justice.services</groupId>
+            <artifactId>aggregate-snapshot-repository-liquibase</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.squareup</groupId>
+            <artifactId>javapoet</artifactId>
```
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/pom.xml]: fixed and corrected to test scope

File: aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/SnapshotAwareEventStreamManager.java [hunk 23]
```diff
@@ -0,0 +1,147 @@
...
+            snapshotService.attemptAggregateStore(id, currentVersion, aggregate, snapshotService.getLatestSnapshotVersion(id, aggregate.getClass()));
+        }
+    }
+
+    private JsonEnvelope eventWithVersion(final JsonEnvelope event, final UUID streamId, final Long version) {
+        final JsonObjectBuilder stream = Json.createObjectBuilder()
+                .add(STREAM_ID, streamId.toString())
+                .add(VERSION, version);
+
+        final JsonObjectBuilder metadata = JsonObjects.createObjectBuilderWithFilter(event.metadata().asJsonObject(), x -> !STREAM.equals(x))
```
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/SnapshotAwareEventStreamManager.java]: Static Import applied

File: aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/SnapshotAwareEventStreamManager.java [hunk 24]
```diff
@@ -0,0 +1,147 @@
...
+                .add(STREAM, stream.build());
+
+        return DefaultJsonEnvelope.envelopeFrom(JsonObjectMetadata.metadataFrom(metadata.build()), event.payloadAsJsonObject());
```
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/main/java/uk/gov/justice/services/eventsourcing/source/core/SnapshotAwareEventStreamManager.java]: Static Import applied

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/domain/aggregate/classloader/DynamicAggregateTestClassGenerator.java (hunk 4 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/domain/aggregate/classloader/DynamicAggregateTestClassGenerator.java]: Finals Applied

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java (hunk 5 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: Static imports applied

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java (hunk 7 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: Commented code removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java (hunk 8 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: Commented code removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java (hunk 9 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: Commented code removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java (hunk 10 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceIT.java]: Commented code removed

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java (hunk 12 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java]: Static Import Applied

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java (hunk 11 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/core/aggregate/SnapshotAwareAggregateServiceTest.java]: Applied consistent finals

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java (hunk 13 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java]: Final applied

File: aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java (hunk 14 above)
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/src/test/java/uk/gov/justice/services/eventsourcing/source/core/snapshot/DefaultSnapshotServiceTest.java]: Final Applied

File: aggregate-snapshot/aggregate-snapshot-service/pom.xml [hunk 25]
```diff
@@ -0,0 +1,157 @@
...
+    <parent>
+        <artifactId>aggregate-snapshot</artifactId>
+        <groupId>uk.gov.justice.services</groupId>
+        <version>0.31.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>aggregate-snapshot-service</artifactId>
+    <dependencies>
+        <dependency>
```
Comment (MohamedFarouk-HMCTS) [inline aggregate-snapshot/aggregate-snapshot-service/pom.xml]: Arranged in the order of main modules first followed by test modules

**Pull request merged**